nomad/structs/services.go

```diff
@@ -254,6 +263,14 @@ func (sc *ServiceCheck) validate() error {
 		return fmt.Errorf("invalid address_mode %q", sc.AddressMode)
 	}
 
+	// Validate OnUpdate
+	switch sc.OnUpdate {
+	case "", OnUpdateIgnore, OnUpdateRequireHealthy, OnUpdateIgnoreWarn:
```

Comment:
`""` isn't actually valid, right? it gets canonicalized to OnUpdateRequireHealthy

Reply:
Actually I think it is valid, I'm not sure if Canonicalize is guaranteed to be called before validate? `AddressMode` does the same thing